components/node/rollup/types.go

```diff
@@ -114,6 +114,20 @@ func (cfg *Config) ValidateL2Config(ctx context.Context, client L2Client) error
 	return nil
 }
 
+func (cfg *Config) TargetBlockNumber(timestamp uint64) (num uint64, err error) {
+	// subtract genesis time from timestamp to get the time elapsed since genesis, and then divide that
+	// difference by the block time to get the expected L2 block number at the current time. If the
+	// unsafe head does not have this block number, then there is a gap in the queue.
+	genesisTimestamp := cfg.Genesis.L2Time
+	if timestamp < genesisTimestamp {
+		return 0, fmt.Errorf("did not reach genesis time (%d) yet", genesisTimestamp)
+	}
+	wallClockGenesisDiff := timestamp - genesisTimestamp
```

Comment:
This variable name seems wrong because when you seeing the https://pkg.go.dev/time#hdr-Monotonic_Clocks `time.Now()` returns monotonous clock rather than wall clock.

```diff
-wallClockGenesisDiff := timestamp - genesisTimestamp
+timeDiff := timestamp - genesisTimestamp
```